lib/matplotlib/tests/test_figure.py

```diff
@@ -66,6 +66,16 @@ def test_figure():
     plt.close('tomorrow')
 
 
+@image_comparison(baseline_images=['figure_legend'])
+def test_figure_legend():
+    fig, axes = plt.subplots(2)
+    axes[0].plot([0, 1], [1, 0], label='x', color='g')
+    axes[0].plot([0, 1], [0, 1], label='y', color='r')
+    axes[1].plot([0, 1], [1, 0], label='y', color='r')
```

Comment:
What happens when the labels are the same, but the style is different? I believe it should be separate, but it's not tested.